### DISCUSSION TITLE
Self reference an action for testing

### DISCUSSION THREAD

**Discussion by jacobtomlinson** [community/community#25194] [Actions]
<p>When writing an action I would like for it to trigger itself as part of the CI process.</p>
<p>I have a step which runs the action, and then another step which tests the output to ensure they are as expected.</p>
<p>I currently have a workflow where actions run their own master branch as part of their CI, but ideally, if someone raises a PR I want the PR version of the action to run.</p>
<p>I’ve tried changing the step to something like this</p>
<pre><code>- name: Self test
       id: selftest
       uses: ${{ github.repository }}@${{ github.ref }}
</code></pre>
<p>But I’m getting the error</p>
<pre><code>- Your workflow file was invalid: The pipeline is not valid. .github/workflows/integration.yml (Line: 10, Col: 15): Unrecognized named-value: 'github'. Located at position 1 within expression: github.repository
</code></pre>
<p>Is there any way for an action to run itself as part of a PR?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/jacobtomlinson">@jacobtomlinson</a> ,</p>
<p>I also tested and tried as your shared demo, and also got the same result: <strong>Unrecognized named-value: ‘github’</strong>.</p>
<pre><code>- name: Self test
  id: selftest
  uses: ${{ github.repository }}@${{ github.ref }}

- name: Self test
  id: selftest
  uses: ${{ github.repository }}@${{ github.sha }}

- name: Self test
  id: selftest
  uses: myOrg/myActionRepos@${{ github.sha }}
</code></pre>
<p>Looks like, when using an action in the workflows, we need to explicitly specify the repository/path and version of the action. It seems does not support using variables to specify the repository/path and version of the action for the ’ <strong>uses</strong>’ section.</p>
<p>More details, you can refernce the docs about the syntax <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsuses" rel="nofollow noopener"><strong>jobs.&lt;job_id&gt;.steps.uses</strong></a>.</p>
<p>If your projects really need this feature, I recommend you directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by vidavidorra** [comment, marked as answer]
<p>You should be able to do so by referencing the local action. E.g. using the followign snippet.</p>
<pre><code>- name: Self test
  id: selftest
  uses: ./
</code></pre>
<p>As reference, see my repo github<a href="https://github.com/vidavidorra/github-action-renovate/blob/cd4dee329720030349b278460f2f6ff8ddaca4c4/.github/workflows/example-basic.yml" rel="nofollow noopener">-action-renovate</a> where I also use this (I haven’t tested it with PRs though, but it should work).</p>
